### PR TITLE
:point_right: Use common CI scripts

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
There are 2 related PR's required to make the common CI scripts work with the connecting to services CI build.

https://github.com/nhsuk/nhsuk-rancher-templates/pull/14
https://github.com/nhsuk/ci-deployment/pull/4

There are some hacks used to allow testing.

- The submodule used by this PR is pointing at the ci-deployment feature branch feature/remove-profile-specificity (instead of master)
- The rancher_pr_deploy.sh script is pulling the docker-compose.yml & rancher-compose.yml files from the feature/changes-for-ci-script branch (instead of master) of the nhsuk/nhsuk-rancher-templates repo

If the changes proposed are OK then these hacks will have to be removed once the changes for each PR have been merged.

There are lots of improvements we can make to the scripts but at this point in time it works.

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by st3v3nhunt** [comment]
Adding WIP label until PR nhsuk/nhsuk-rancher-templates#14 is complete.
See https://github.com/nhsuk/nhsuk-rancher-templates/pull/14#issuecomment-279985585 for additional info

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
test

**Comment by c2s-dev** [comment]
test

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://connecting-to-services-pr-260.dev.c2s.nhschoices.net](http://connecting-to-services-pr-260.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by c2s-dev** [comment]
:warning: deployment of 260 for nhsuk/connecting-to-services failed

**Comment by neilbmclaughlin** [comment]
Awaiting merge of work in #267 to fix snyk vulnerability

File: .travis.yml [hunk 1]
```diff
@@ -3,14 +3,24 @@ sudo: required
 services:
   - docker
 
+env:
+  global:
+  - DOCKER_REPO=nhsuk/connecting-to-services
+  - DOCKER_USERNAME=nhsukautomata
+  - RANCHER_ACCESS_KEY=655696CC96424E6D4DB0
```
Comment (st3v3nhunt) [inline .travis.yml]: This should be removed and managed by Travis (which it already is)

File: .travis.yml [hunk 2]
```diff
@@ -3,14 +3,24 @@ sudo: required
...
+  - RANCHER_ENVIRONMENT=c2s-dev
```
Comment (st3v3nhunt) [inline .travis.yml]: This should be removed and managed by Travis (which it already is)

**Comment by st3v3nhunt** [comment]
The Snyk failure gives us an interesting situation. PR #267 won't have a successful build till these changes are included. And this PR wont successfully run all checks until #267 has been merged.
Perhaps these should be merged?

**Comment by st3v3nhunt** [comment]
Based on the [comment above](https://github.com/nhsuk/connecting-to-services/pull/260#issuecomment-286913924), I've combined the changes from PR #267 and this one to get to a point where everything that needs to be done to get a passing build is in once place. that place being PR #269 
If that works it will supersede this one and this can be closed.

**Comment by st3v3nhunt** [comment]
PR #269 is looking pretty good from the POV of the build checks being successful and the deployment happening. Closing this.

**Pull request closed without merging**